### PR TITLE
build.bat: Add static and shared options

### PR DESCRIPTION
**Pull request by 1480c1**
Whichever one comes last will be the one that is built, Should help with people who want to choose their build type.

### CODE REVIEW

File: Build/windows/build.bat [hunk 1]
```diff
@@ -119,6 +126,6 @@ goto :args
 
 :help
     echo Batch file to build SVT-AV1 on Windows
-    echo Usage: generate.bat 2019^|2017^|2015^|clean [release|debug] [nobuild] [test]
+    echo Usage: generate.bat 2019^|2017^|2015^|clean [release|debug] [nobuild] [test] [shared|static]
```
Comment (tianjunwork) [inline Build/windows/build.bat:129]: generate.bat ==> build.bat
2019^|2017^|2015^|clean ==>[2019^|2017^|2015^|clean]

Comment (1480c1) [reply Build/windows/build.bat]: Seems I forgot to escape some stuff

**Comment by ePirat** [comment]
Does it fix the buildtype as well? Last time I tried to use that it just errored out with a confusing error so I had to use cmake directly.

**Comment by 1480c1** [comment]
It at least does for me, although it doesn't properly do it with non msvc generators since that requires setting `-DCMAKE_BUILD_TYPE=<type>`, but it wouldn't be good to add that without checking the generator, but bat doesn't have a good way to do case switches

File: Build/windows/build.bat [hunk 2]
```diff
@@ -4,24 +4,41 @@
...
 setlocal
 cd /d "%~dp0"
-set instdir=%CD%
 
+:: Set defaults to prevent inheriting
 set "build=y"
+:: Default is debug
+set "buildtype=Debug"
+:: Default is shared
+set "static=ON"
```
Comment (tianjunwork) [inline Build/windows/build.bat:13]: `static=on` means building shared? not easy to read.

Comment (1480c1) [reply Build/windows/build.bat:13]: Should I change this to `shared` instead?

Comment (1480c1) [reply Build/windows/build.bat]: Changed it

**Comment by tianjunwork** [comment]
Hi @ePirat , could you also review and approve? Thank you.

**Pull request merged**